### PR TITLE
Separate tabs for Downloaded and Published files in My Files

### PR DESCRIPTION
**Pull request by alexliebowitz**
Depends on [this](https://github.com/lbryio/lbry-web-ui/pull/57) lbry-web-ui PR and [this](https://github.com/lbryio/lbry/pull/165) lbrynet PR.

### CODE REVIEW

File: js/page/my_files.js [hunk 1]
```diff
@@ -131,34 +132,100 @@ var MyFilesRow = React.createClass({
...
+          let name = fileInfo.lbry_uri;
+
+          lbry.search(name, (results) => {
+            var result = results[0];
+
+            if (result.name != name) {
+              // File not listed in Lighthouse
+              var available = false;
+            } else {
+              var available = result.available;
```
Comment (kauffj) [inline js/page/my_files.js]: var available = result.name == name && result.available

File: js/page/my_files.js [hunk 2]
```diff
@@ -131,34 +132,100 @@ var MyFilesRow = React.createClass({
...
+        filesInfo: filesInfo,
       });
-      this.fileTimeout = setTimeout(() => { this.updateFilesInfo() }, 1000);
+
+      this._fileTimeout = setTimeout(() => { this.updateFilesInfo() }, 1000);
     });
   },
   render: function() {
-    if (this.state.filesInfo === null) {
+    if (this.state.filesInfo === null || !this._filesOwnershipLoaded) {
```
Comment (kauffj) [inline js/page/my_files.js]: Can `this._filesOwnershipLoaded` instead be a state variable? And the callback from `componentWillMount` would trigger `setState`.

File: js/lbry.js [hunk 3]
```diff
@@ -169,14 +173,42 @@ lbry.revealFile = function(path, callback) {
...
+  // If timeoutCallback is provided, it will be called if the file fails to appear.
+  lbry.getFilesInfo(function(filesInfo) {
+    for (var fileInfo of filesInfo) {
+      if (fileInfo.lbry_uri == name) {
+        callback(fileInfo);
+        return;
+      }
+    }
+
+    if (tryNum <= 200) {
```
Comment (kauffj) [inline js/lbry.js]: This code makes me think we need a long-polling and/or daemon pushing to registered interface solution.

**Comment by kauffj** [comment]
@alexliebowitz long-polling not necessary before a merge, just a component. If I am wrong about `setState` pattern in some way, you can just merge, if I am correct, please fix and then merge.

File: js/page/my_files.js (hunk 2 above)
Comment (alexliebowitz) [inline js/page/my_files.js]: This hit me as I was falling asleep after committing this, I'll do it.

File: js/lbry.js (hunk 3 above)
Comment (alexliebowitz) [inline js/lbry.js]: Me and Jack have talked about that. It's tricky but possible. There's no standard way to do push or long polling in JSON-RPC over HTTP. You can have the server make RPC calls to the client, but there's no way to listen for connections from the browser. We might be able to do it with WebSockets. JSON-RPC supports arbitrary transport layers, but I'm not sure if the Python library we're using does.

Comment (kauffj) [inline js/lbry.js]: I think WebSockets can work here, but def. not worth blocking on. Fix the other thing and ship it.

**Pull request merged**